messaging/WFLY-13793_disable_legacy_amq1_prefix.adoc

```diff
@@ -0,0 +1,128 @@
+
+* For a destination (queue or topic) created by annotation `@JMSDestinationDefinition` on a remote Artemis Server, add a `enable-amq1-prefix` boolean property to control the enablement of Artemis 1.x prefix. If the property is not defined, the default is `true` (meaning that name retro-compatibility is enabled).
+
+[source, java]
+----
+@JMSDestinationDefinition(
+    // explicitly mention a resourceAdapter corresponding to a pooled-connection-factory resource to the remote server
+    resourceAdapter = "remote-artemis",
+    name="java:global/env/myQueue2",
+    interfaceName="javax.jms.Queue",
+    destinationName="myQueue2",
+    properties = {
+        "enable-amq1-prefix=false"
+    }
+)
+----
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+== Test Plan
+
+* WildFly integration testsuite `ExternalJMSDestinationManagementTestCase`
+
+Prepare Wildfly server messaging-activemq subsystem be able to use internal broker as remote one.
+Create queue `myExternalQueue` and `myExternalTopic` with `enable-amq1-prefix` set to `false`. Both created queues name should not contain amq1 prefix `jms.queue` or `jms.topic`.
+
+The opposite version of the test is: create queue `myExternalLegacyQueue` and `myExternalLegacyTopic` with `enable-amq1-prefix` set to `true`. Both created queue contains amq1 prefix `jms.queue` or `jms.topic`.
+
+
+
+* WildFly integration testsuite `ExternalJMSDestinationDefinitionMessagingDeploymentTestCase`
+
+Prepare Wildfly server messaging-activemq subsystem be able to use internal broker as remote one.
+Create an external JMS queue on a remote broker with the amq1 prefix disabled through JakartaEE annotation `@JMSDestinationDefinition`. i.e.:
+
+[source, java]
+----
+@JMSDestinationDefinitions(
+        value = {
+            @JMSDestinationDefinition(
+                    resourceAdapter = REMOTE_PCF,
+                    name = QUEUE_LOOKUP,
+                    interfaceName = "javax.jms.Queue",
+                    destinationName = QUEUE_NAME,
+                    properties = {"enable-amq1-prefix=false"}
+            ),
+            @JMSDestinationDefinition(
+                    resourceAdapter = REMOTE_PCF,
+                    name = TOPIC_LOOKUP,
+                    interfaceName = "javax.jms.Topic",
+                    destinationName = TOPIC_NAME,
+                    properties = {"enable-amq1-prefix=false"}
+            )
+        }
+)
+----
+
+It's expected to create a queue without amq1 prefix `jms.queue` or `jms.topic`, and be able to produce messages on the queue and receive same messages with MDB listening on that queue name without amq1 prefix `jms.queue` or `jms.topic`.
```

Comment:
@soul2zimate we also need a test to check what happens when both `messaging-activemq.pooled-connection-factory.enable-amq1-prefix` and `@JMSDestinationDefinition.enable-amq1-prefix` are set and they have different values

Reply:
@soul2zimate same as above: after creating the destination we should also check we can send/receive from that very same destination e.g. we create `MyQueue` and then we send/receive from `MyQueue` (not `jms.queue.MyQueue`)